joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/passes/PackageCtorCreationPass.scala

```diff
@@ -11,12 +11,16 @@ import scala.jdk.CollectionConverters.*
 
 class PackageCtorCreationPass(cpg: Cpg, config: Config, goGlobal: GoGlobal)
     extends ConcurrentWriterCpgPass[(String, Set[(Ast, String)])](cpg) {
-  override def generateParts(): Array[(String, Set[(Ast, String)])] =
-    goGlobal.pkgLevelVarAndConstantAstMap
+  override def generateParts(): Array[(String, Set[(Ast, String)])] = {
+    val parts = goGlobal.pkgLevelVarAndConstantAstMap
```

Comment:
Could also use a `try...finally` to ensure this clears and not require a variable assignment.

For clarity, might help to make a case class for the `(String, Set[(Ast, String)])` types

Reply:
done